config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,210 @@
+              # Wait for all processes to complete
+              wait $p1 $p2 $p3
+              wait $p1 && wait $p2 && wait $p3
+      startupProbe:
+        failureThreshold: 40
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              ray status
+  volumes:
+    - name: shm
+      emptyDir:
+        medium: Memory
+        sizeLimit: 12Gi
+  workerSpec:
+    containers:
+      - name: worker-container
+        image: kserve/huggingfaceserver:latest
+        command: ["bash", "-c"]
+        args:
+          - |
+            SECONDS=0
+
+            while true; do              
+              if (( SECONDS <= 120 )); then
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379" > /dev/null 2>&1; then
+                  echo "GCS is ready."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Waiting for GCS to be ready."
+              else
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                  echo "GCS is ready. Any error messages above can be safely ignored."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Still waiting for GCS to be ready."
+                echo "For troubleshooting, refer to the FAQ at https://github.com/ray-project/kuberay/blob/master/docs/guidance/FAQ.md."
+              fi
+              
+              sleep 5
+            done
+
+            echo "Attempting to connect to Ray cluster at $RAY_HEAD_ADDRESS ..."
+            RAY_HEAD_ADDRESS="${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"
+            ray start --address="$RAY_HEAD_ADDRESS" --block
+        env:
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace              
+        resources:
+          limits:
+            cpu: "16"
+            memory: 48Gi
+          requests:
+            cpu: "8"
+        volumeMounts:
+          - name: shm
+            mountPath: /dev/shm
+        livenessProbe:
+          failureThreshold: 3
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 5
+          exec:
+            command:
+              - bash
+              - -c
+              - |
+                [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]
+        startupProbe:
+          failureThreshold: 12
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 5
+          exec:
+            command:
+              - ray
+              - status
+    volumes:
+      - name: shm
+        emptyDir:
+          medium: Memory
+          sizeLimit: 12Gi
```

Comment:
what is the purpose of this?

Reply:
This shared memory is needed for a Ray cluster, especially when running tasks that involve large data structures or inter-process communication (IPC). When using multi-node or multi-GPU inference (like serving very large language models in distributed Ray clusters), loading models into shared memory can speed up operations by allowing multiple processes or containers to access model weights and data concurrently, without extra memory overhead.

However, this value is not yet fully optimized, and it is based on research findings. For a 70B model split across 3 nodes (with 1 GPU per node), approximately 8-12 GiB of shared memory per node might be required.

Reply:
okay, this would be a good information to be added in the docs as well.

Reply:
This is the same. I am not sure it would be a good idea to add this to the doc because it is a kind of assumption. @yuzisun has a plan to do a performance test so we can wait for his result :)

Reply:
If you have only one 1 gpu per node, does it need shared memory? Also, isn't it better for the users to override it in the inference service spec as it is going to vary across models(based on number of gpus)?

Reply:
From my understanding, it can be overwritten by isvc spec but I need to do double-chek.

Reply:
Shard memory is only needed for high tensor parallel lism for cross gpu communication on the same node

Reply:
@yuzisun good point. for multi-node cases, do you think we can remove this shared memory?